packages/web-pkg/src/components/AvatarUpload.vue

```diff
@@ -0,0 +1,210 @@
+<template>
+  <div class="avatar-upload">
+    <input
+      ref="fileInputRef"
+      class="oc-invisible"
+      type="file"
+      accept="image/jpeg, image/png"
+      @change="onFileChange"
+    />
+    <div class="oc-flex oc-flex-column oc-flex-middle">
+      <oc-avatar
+        class="oc-mb-m"
+        :width="128"
+        :userid="user.onPremisesSamAccountName"
+        :user-name="user.displayName"
+        :src="userAvatar"
+      />
+      <div>
+        <div class="oc-button-group">
+          <oc-button size="small" @click="triggerFileInput">
+            {{ $gettext('Upload') }}
+          </oc-button>
+          <oc-button v-if="userAvatar" size="small" @click="showRemoveModal = true">
+            {{ $gettext('Remove') }}
+          </oc-button>
+        </div>
+      </div>
+    </div>
+    <oc-modal
+      v-if="showCropModal"
+      :title="$gettext('Crop your new profile picture')"
+      :button-cancel-text="$gettext('Cancel')"
+      :button-confirm-text="$gettext('Set')"
+      :button-confirm-disabled="!cropperReady"
+      @cancel="onCropModalCancel"
+      @confirm="onCropModalConfirm"
+    >
+      <template #content>
+        <div v-if="imageUrl">
+          <img ref="imageRef" class="avatar-upload-modal-image" :src="imageUrl" />
+        </div>
+      </template>
+    </oc-modal>
+    <oc-modal
+      v-if="showRemoveModal"
+      :message="$gettext('Are you sure you want to remove your profile picture?')"
+      :title="$gettext('Remove profile picture')"
+      :button-cancel-text="$gettext('Cancel')"
+      :button-confirm-text="$gettext('Remove')"
+      @cancel="showRemoveModal = false"
+      @confirm="onRemoveModalConfirm"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, watch, nextTick } from 'vue'
+import Cropper from 'cropperjs'
+import 'cropperjs/dist/cropper.css'
+import { useAvatarsStore, useClientService, useMessages, useUserStore } from '../composables'
+import { storeToRefs } from 'pinia'
+import { useGettext } from 'vue3-gettext'
+
+const userStore = useUserStore()
+const avatarsStore = useAvatarsStore()
+const { user } = storeToRefs(userStore)
+const { userAvatar } = storeToRefs(avatarsStore)
+
+const { $gettext } = useGettext()
+const { showErrorMessage, showMessage } = useMessages()
+const { graphAuthenticated } = useClientService()
+
+const imageUrl = ref<string | null>(null)
+const imageRef = ref<HTMLImageElement | null>(null)
+const cropper = ref<Cropper | null>(null)
+const cropperReady = ref(false)
+const showCropModal = ref(false)
+const showRemoveModal = ref(false)
+const fileInputRef = ref<HTMLInputElement | null>(null)
+const maxFileSize = 10 * 1024 * 1024 // 10MB
+
+const onFileChange = (event: Event) => {
+  const target = event.target as HTMLInputElement
+  const file = target.files?.[0]
+  if (!file) return
+
+  if (file.size > maxFileSize) {
+    showErrorMessage({ title: $gettext('File size exceeds the limit of 10MB') })
```

Comment:
Please use gettext interpolation for the `10`. You can assign the `10` to a differrent variable before line 80 and then use it for the calculation of the `maxFileSize` variable and for the gettext interpolation. Would be awful if we'd need to change translations when we decide to change the file size limit.